### ISSUE TITLE
Refactor User into Messages

### ISSUE THREAD

**Issue by gempir** [gempir/go-twitch-irc#83]
I'm thinking about refactoring the User struct into the messages themselves. This would mean the API for the handler could only be `func(Message)` and this would make the AddHandler function we are considering borrowed from discordgo syntax. 

This would also make it more obvious which messages have User information, and which don't.

Any opinions on this?

**Comment by FletcherCodes** [comment]
While I prefer the explicitness of the separate User and Message structs, I can see the benefits of the AddHandler function. From taking a look at some of the discordgo examples, it'd be significantly cleaner on the user end.

Will this also mean multiple handlers for each event?

**Comment by gempir** [comment]
I don't think I understand the question. Could you rephrase?

If you mean multiple client added handlers for the same event then yea sure we could do that.

**Comment by FletcherCodes** [comment]
Yeah, that's exactly what I mean.
